homeassistant/components/light/__init__.py

```diff
@@ -241,7 +241,20 @@ def async_handle_light_service(service):
         if color_name is not None:
             params[ATTR_RGB_COLOR] = color_util.color_name_to_rgb(color_name)
 
+        xy_color = params.get(ATTR_XY_COLOR, None)
+
         for light in target_lights:
+            # Convert xy_color if unsupported
+            if xy_color is not None:
+                support_xy = bool(light.supported_features & SUPPORT_XY_COLOR)
+                brightness = params.get(ATTR_BRIGHTNESS, light.brightness)
```

Comment:
I actually thought this to be an improvement over the OSRAM case where current brightness was used always, not just as a fallback. Apart from lag due to polling, I do not understand how it could be wrong, can you elaborate?

Most importantly: what can I use for the current brightness when an ```xy_color``` is set without a ```brightness```?